aiolifx/__main__.py

```diff
@@ -62,315 +66,345 @@ def unregister(self, bulb):
-                                    [
-                                        58275,
-                                        0,
-                                        int(round((float(lov[1]) * 65365.0) / 100.0)),
-                                        int(round(float(lov[2]))),
-                                    ]
-                                )
 
-                                MyBulbs.boi = None
-                            except:
-                                print(
-                                    "Error: For white brightness (0-100) and temperature (2500-9000) must be numbers.\n"
-                                )
-                        else:
-                            print(
-                                "Error: For white you must indicate brightness (0-100) and temperature (2500-9000)\n"
-                            )
-                    elif int(lov[0]) == 3:
-                        if len(lov) > 3:
-                            try:
-                                MyBulbs.boi.set_color(
-                                    [
-                                        int(round((float(lov[1]) * 65535.0) / 360.0)),
-                                        int(round((float(lov[2]) * 65535.0) / 100.0)),
-                                        int(round((float(lov[3]) * 65535.0) / 100.0)),
-                                        3500,
-                                    ]
-                                )
-                                MyBulbs.boi = None
-                            except:
-                                print(
-                                    "Error: For colour hue (0-360), saturation (0-100) and brightness (0-100)) must be numbers.\n"
-                                )
-                        else:
-                            print(
-                                "Error: For colour you must indicate hue (0-360), saturation (0-100) and brightness (0-100))\n"
-                            )
+def get_features(bulb):
+    base_options = [
+        DeviceFeatures.INFO,
+        DeviceFeatures.FIRMWARE,
+        DeviceFeatures.WIFI,
+        DeviceFeatures.UPTIME,
+        DeviceFeatures.REBOOT,
+    ]
+    features = []
+    if alix.aiolifx.products_dict[bulb].max_kelvin != None:
+        features.extend([DeviceFeatures.POWER, DeviceFeatures.WHITE])
+    if alix.aiolifx.products_dict[bulb].color is True:
+        features.extend([DeviceFeatures.COLOR, DeviceFeatures.PULSE])
+    if alix.aiolifx.products_dict[bulb].hev is True:
+        features.extend([DeviceFeatures.HEV_CYCLE, DeviceFeatures.HEV_CONFIGURATION])
+    if alix.aiolifx.products_dict[bulb].multizone is True:
+        features.extend(
+            [DeviceFeatures.FIRMWARE_EFFECT, DeviceFeatures.FIRMWARE_EFFECT_START_STOP]
+        )
+    if alix.aiolifx.products_dict[bulb].buttons is True:
+        features.append(DeviceFeatures.RELAYS)
+    features.extend(base_options)
+    return features
```

Comment:
This logic should be contained in the `devide` itself once we support different device classes: https://github.com/frawau/aiolifx/issues/69